### PR TITLE
[REVIEW] Convert groupby CFFI to Cython

### PR DESCRIPTION
**Pull request by kkraus14**
Depends on #1479

### CODE REVIEW

File: python/cudf/bindings/groupby.pyx [hunk 1]
```diff
@@ -0,0 +1,462 @@
...
+    in columns using the naming scheme of `aggregation_columnname`.
+    """
+    result = DataFrame()
+    add_col_values = True
+
+    cdef gdf_context* ctx = create_context_view(0, groupby_class._method, 0, 0, 0)
+
+    sort_results = True
+
+    # TODO: Use MultiColumn here instead of use_prefix
```
Comment (dantegd) [inline python/cudf/bindings/groupby.pyx]: Would be good to create a github issue and/or link it here for tracking

File: python/cudf/bindings/groupby.pyx [hunk 2]
```diff
@@ -0,0 +1,462 @@
...
+                            col_agg.size,
+                            dtype=groupby_class._df[val_col]._column.data.dtype
+                        )
+                    )
+                )
+
+        out_col_agg = column_view_from_column(out_col_agg_series._column)
+
+        if agg_type is None:
+            raise RuntimeError(
```
Comment (dantegd) [inline python/cudf/bindings/groupby.pyx]: Should be a `NotImplementedError`

File: python/cudf/bindings/groupby.pyx [hunk 3]
```diff
@@ -0,0 +1,462 @@
...
+                    nvcat_obj = nvcategory.bind_cpointer(nvcat_ptr)
+                    nvstr_obj = nvcat_obj.to_strings()
+                else:
+                    import nvstrings
+                    nvstr_obj = nvstrings.to_device([])
+                out_col_values_series[i]._column._data = nvstr_obj
+                out_col_values_series[i]._column._nvcategory = nvcat_obj
+        if out_col_agg_series.dtype == np.dtype("object") and \
+                len(out_col_agg_series) > 0:
+            import nvcategory
```
Comment (dantegd) [inline python/cudf/bindings/groupby.pyx]: Isn't this the exact same code block as above (lines 227 to 238)? If so maybe try making them a function?

File: python/cudf/groupby/groupby.py [hunk 4]
```diff
@@ -1,21 +1,13 @@
...
-from cudf.dataframe.buffer import Buffer
-from cudf.dataframe.categorical import CategoricalColumn
-from cudf.utils.cudautils import zeros
-from cudf.bindings.nvtx import nvtx_range_pop
-import cudf.dataframe.index as index
-
-from libgdf_cffi import ffi, libgdf
-from librmm_cffi import librmm as rmm
+from cudf.bindings.groupby import (
+    agg as cpp_agg,
```
Comment (dantegd) [inline python/cudf/groupby/groupby.py]: Any reason to have underscode in `_cpp_apply_basic_agg` but not in `cpp_agg`?

Comment (kkraus14) [reply python/cudf/groupby/groupby.py]: I just copied what existed, previously. I wasn't trying to remake the wheel 😄.

File: python/cudf/bindings/groupby.pyx (hunk 3 above)
Comment (kkraus14) [reply python/cudf/bindings/groupby.pyx]: @dantegd this is fixed in #1492

**Pull request merged**